clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -300,14 +297,6 @@ private RecordAppendResult tryAppend(long timestamp, byte[] key, byte[] value, H
                 }
             }
         }
-        if (!expiredBatches.isEmpty()) {
-            log.trace("Expired {} batches in accumulator", count);
```

Comment:
Maybe we can move this log line into `Sender` after we have collected the expired batches.